pkg/controller/master/rancher/register.go

```diff
@@ -110,75 +114,90 @@ func (h *Handler) registerExposeService() error {
 				},
-				Ports: []corev1.ServicePort{
-					{
-						Name:       "https-internal",
-						Port:       443,
-						Protocol:   corev1.ProtocolTCP,
-						TargetPort: intstr.FromInt(443),
-					},
-					{
-						Name:       "http",
-						Port:       80,
-						Protocol:   corev1.ProtocolTCP,
-						TargetPort: intstr.FromInt(80),
-					},
+				{
+					Name:       "http",
+					Port:       80,
+					Protocol:   corev1.ProtocolTCP,
+					TargetPort: intstr.FromInt(80),
 				},
 			},
-		}
+		},
+	}
 
-		// set vip loadBalancer type and ip
-		enabled, err := strconv.ParseBool(vip.Enabled)
-		if err != nil {
-			return err
-		}
-		if enabled && vip.ServiceType == corev1.ServiceTypeLoadBalancer {
-			svc.Spec.Type = vip.ServiceType
-			// After kube-vip v0.5.2, it uses annotation kube-vip.io/loadbalancerIPs to set the loadBalancerIP
-			if strings.ToLower(vip.Mode) == vipDHCPMode {
-				svc.Annotations[keyKubevipRequestIP] = vip.IP
-				svc.Annotations[keyKubevipHwaddr] = vip.HwAddress
-				svc.Annotations[keyKubevipLoadBalancerIPs] = vipDHCPLoadBalancerIP
-			} else {
-				svc.Annotations[keyKubevipLoadBalancerIPs] = vip.IP
-			}
+	// set vip loadBalancer type and ip
+	enabled, err := strconv.ParseBool(vip.Enabled)
+	if err != nil {
+		return err
+	}
+	if enabled && vip.ServiceType == corev1.ServiceTypeLoadBalancer {
+		svc.Spec.Type = vip.ServiceType
+		// After kube-vip v0.5.2, it uses annotation kube-vip.io/loadbalancerIPs to set the loadBalancerIP
+		if strings.ToLower(vip.Mode) == vipDHCPMode {
+			svc.Annotations[keyKubevipRequestIP] = vip.IP
+			svc.Annotations[keyKubevipHwaddr] = vip.HwAddress
+			svc.Annotations[keyKubevipLoadBalancerIPs] = vipDHCPLoadBalancerIP
+		} else {
+			svc.Annotations[keyKubevipLoadBalancerIPs] = vip.IP
 		}
+	}
 
-		if _, err := h.Services.Create(svc); err != nil {
-			return err
-		}
+	if _, err := h.Services.Create(svc); err != nil {
+		return err
 	}
-	// Update annotations of the ingress-expose service created before Harvester v1.2.0
-	if svc.Annotations == nil || svc.Annotations[keyKubevipIgnoreServiceSecurity] != trueStr {
-		svcCopy := svc.DeepCopy()
-		if svc.Annotations == nil {
-			svcCopy.Annotations = make(map[string]string)
-		}
-		svcCopy.Annotations[keyKubevipIgnoreServiceSecurity] = trueStr
-		if _, ok := svcCopy.Annotations[keyKubevipLoadBalancerIPs]; !ok {
-			svcCopy.Annotations[keyKubevipLoadBalancerIPs] = svc.Spec.LoadBalancerIP
-		}
-		if _, err := h.Services.Update(svcCopy); err != nil {
-			return err
-		}
+
+	return nil
+}
+
+func (h *Handler) updateIngressExposeService(svc *corev1.Service) error {
+	svcCopy := svc.DeepCopy()
+	if svc.Annotations == nil {
+		svcCopy.Annotations = make(map[string]string)
+	}
+	svcCopy.Annotations[keyKubevipIgnoreServiceSecurity] = trueStr
+	if _, ok := svcCopy.Annotations[keyKubevipLoadBalancerIPs]; !ok {
+		svcCopy.Annotations[keyKubevipLoadBalancerIPs] = svc.Spec.LoadBalancerIP
+	}
+	if _, err := h.Services.Update(svcCopy); err != nil {
+		return err
```

Comment:
when `Services.Update` is successful but `restartKubevipPod` is failed, seems `restartKubevipPod` has no chance to be rescheduled

Reply:
Yes, considering the time limit, the PR is just a workaround. I'll submit a PR to kube-vip to address the problem thoroughly.